discord/interactions.py

```diff
@@ -178,15 +178,18 @@ def _from_data(self, data: InteractionPayload):
 
         # TODO: there's a potential data loss here
         if self.guild_id:
-            guild = self.guild or Object(id=self.guild_id)
+            guild = self.guild or self._state._get_guild(self.guild_id)
             try:
                 member = data["member"]  # type: ignore
             except KeyError:
                 pass
             else:
-                cache_flag = self._state.member_cache_flags.interaction
-                self.user = guild._get_and_update_member(member, int(member["user"]["id"]), cache_flag)
                 self._permissions = int(member.get("permissions", 0))
+                if guild:
```

Comment:
```suggestion
                if isinstance(guild, Object):
```

Reply:
I believe this should be the opposite? i.e `not isinstance` or `isinstance(guild, Guild)`

Reply:
> I believe this should be the opposite? i.e `not isinstance` or `isinstance(guild, Guild)`

already edited :)